packages/block-library/src/navigation-link/edit.js

```diff
@@ -329,10 +329,7 @@ export default function NavigationLinkEdit( {
 	} = getColors( context, ! isTopLevelLink );
 
 	function onKeyDown( event ) {
-		if (
-			isKeyboardEvent.primary( event, 'k' ) ||
-			( ( ! url || isDraft || isInvalid ) && event.keyCode === ENTER )
```

Comment:
This is handled by the `onClick` event that gets added later on. It uses the same checks.